src/App.js

```diff
@@ -1,16 +1,48 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import { useState } from 'react';
+import ChatLog from './components/ChatLog.js';
+// import InfoBar from './components/InfoBar.js';
 
 const App = () => {
+  const [entries, setEntries] = useState(chatMessages);
+  const [hearts, setHearts] = useState(0);
+
+  const updateLikes = (id) => {
+    const updatedEntries = entries.map((entry) => {
+      if (entry.id === id) {
+        const updatedEntry = {...entry};
+        updatedEntry.liked = !updatedEntry.liked;
+        if (updatedEntry.liked) {
+          setHearts(hearts+1);
+        } else {
+          setHearts(hearts-1);
+        }
+        return updatedEntry;
+      } else {
+        return entry;
+      }
+    })
+    setEntries(updatedEntries);
+  }
+
+  let uniqueSendersSet = new Set();
+
+  for (const entry of entries) {
+    uniqueSendersSet.add(entry.sender);
+  }
+
+  const uniqueSendersArr = Array.from(uniqueSendersSet)
```

Comment:
Ah! I love your set of a set here! ✨ ✨ Great thinking on how to make this code handle more than two senders. I'm excited for your future group chat app. 👀